marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -203,71 +199,52 @@ private FailedRequest extractErrorFields(Response response) {
 
   @Override
   public void connect(String host, int port, String database, SecurityContext securityContext){
-    SSLContext sslContext = null;
-    SSLHostnameVerifier sslVerifier = null;
-    X509TrustManager trustManager = null;
 
-    if (host == null)
-    	throw new IllegalArgumentException("No host provided");
+	  if (host == null)
+		  throw new IllegalArgumentException("No host provided");
+	  if (securityContext == null)
+		  throw new IllegalArgumentException("No security context provided");
+
+	  OkHttpClient.Builder clientBuilder = newClientBuilder();
+	  AuthenticationConfigurer authenticationConfigurer = null;
+
+	  // As of 6.1.0, kerberos/saml/certificate are still coded within this class to avoid potential breaks from
+	  // refactoring. Once the tests for these auth methods are running properly, the code for each can be
+	  // safely refactored.
+	  if (securityContext instanceof BasicAuthContext) {
+		  authenticationConfigurer = new BasicAuthenticationConfigurer();
+		  checkFirstRequest = false;
+	  } else if (securityContext instanceof DigestAuthContext) {
+		  authenticationConfigurer = new DigestAuthenticationConfigurer();
+		  checkFirstRequest = true;
+	  } else if (securityContext instanceof KerberosAuthContext) {
+		  configureKerberosAuth((KerberosAuthContext) securityContext, host, clientBuilder);
+		  checkFirstRequest = false;
+	  } else if (securityContext instanceof CertificateAuthContext) {
+		  checkFirstRequest = false;
+	  } else if (securityContext instanceof SAMLAuthContext) {
+		  configureSAMLAuth((SAMLAuthContext) securityContext, clientBuilder);
+		  checkFirstRequest = false;
+	  } else {
+		  throw new IllegalArgumentException("Unsupported security context: " + securityContext.getClass());
+	  }
 
-    OkHttpClient.Builder clientBldr = newClientBuilder();
+	  if (authenticationConfigurer != null) {
+		  authenticationConfigurer.configureAuthentication(clientBuilder, securityContext);
+	  }
 
-	if (securityContext instanceof BasicAuthContext) {
-	    BasicAuthContext basicContext = (BasicAuthContext) securityContext;
-	    if (basicContext.getSSLContext() != null) {
-            sslContext = basicContext.getSSLContext();
```

Comment:
All of this null-checking on sslContext/trustManager was unnecessary, so I removed it. configureSocketFactory already checks to see if these are not null before doing anything with them.